server/src/services/data_queries.service.ts

```diff
@@ -106,18 +106,23 @@ export class DataQueriesService {
     let result;
 
     try {
-      return await service.run(sourceOptions, parsedQueryOptions, dataSource.id, dataSource.updatedAt);
+      return await service.run(sourceOptions, parsedQueryOptions, dataSource.id, dataSource.updatedAt, { user });
```

Comment:
user will have sensitive info like password, shall we pick and pass details which are required, better if we can create a type for user